src/benchq/resource_estimation/graph/graph_estimator.py

```diff
@@ -22,13 +22,16 @@
     GraphResourceInfo,
 )
 from ..magic_state_distillation import Widget, default_widget_list
+from .transformers import remove_isolated_nodes
 
 INITIAL_SYNTHESIS_ACCURACY = 0.0001
 
 
 def substrate_scheduler(graph: nx.Graph, preset: str) -> TwoRowSubstrateScheduler:
```

Comment:
I understand correctly, the removal of `graph.copy` means that this function will mutate the `graph` argument. I think we want to avoid mutating arguments though since this could make it hard for people to figure out what's going on.

Also, do we still need to have the logic removing isolated nodes be in this function now that there is a `remove_isolated_nodes` transformer?

Reply:
I have re-included the `copy` function. Just to make sure that we are not unduly changing the value of the GraphPartition object.

I have modified the transformer so that the code can be re-used.